src/Elmish.WPF/Logging.fs

```diff
@@ -0,0 +1,130 @@
+﻿module Elmish.WPF.Logging
+
+
+let internal compositeLogger loggers s =
+  loggers |> Seq.iter (fun logger -> logger s)
+
+
+type internal PropertyChangedData =
+  { PropertyNameChain: string
+    PropertyName: string }
+
+type internal ValidationErrorsChangedData =
+  { PropertyNameChain: string
+    PropertyName: string }  // TODO: add Errors
+
+type internal TimingData =
+  { PropertyNameChain: string
+    PropertyName: string
+    BindingDataFunctionName: string
+    ElapsedMilliseconds: int64 }
+
+type internal CreatingHiddenWindowData =
+  { PropertyNameChain: string }
+
+type internal CreatingVisibleWindowData =
+  { PropertyNameChain: string }
+
+type internal InitializingBindingsData =
+  { PropertyNameChain: string }
+
+type internal ClosingWindowData =
+  { PropertyNameChain: string }
+
+type internal HindingWindowData =
+  { PropertyNameChain: string }
+
+type internal ShowingHiddenWindow =
+  { PropertyNameChain: string }
+
+type internal NewSubModelSelectedItemSelectionData =
+  { PropertyNameChain: string
+    NewSelection: obj voption }
+
+type internal TryGetMemberCalledData =
+  { PropertyNameChain: string
+    PropertyName: string }
+
+type internal TrySetMemberCalledData =
+  { PropertyNameChain: string
+    PropertyName: string }
+
+type internal GettingErrorsData =
+  { PropertyNameChain: string
+    PropertyName: string option }
+
+
+type internal TraceLogData =
+  | PropertyChangedData of PropertyChangedData
```

Comment:
I suggest removing the `Data` suffix for all DU cases (you can keep it for the type names)